pkg/controller/servicediscovery/servicediscovery_controller.go

```diff
@@ -361,25 +363,34 @@ func updateDNSConfigMap(client controllerClient.Client, k8sclientSet *clientset.
 		}
 		*/
 		corefile := configMap.Data["Corefile"]
-		if strings.Contains(corefile, "lighthouse") {
-			// Assume this means we've already set the ConfigMap up
-			return nil
-		}
 		lighthouseDnsService := &corev1.Service{}
 		err = client.Get(context.TODO(), types.NamespacedName{Name: lighthouseCoreDNSName, Namespace: cr.Namespace}, lighthouseDnsService)
-		if err != nil || lighthouseDnsService.Spec.ClusterIP == "" {
+		lighthouseClusterIp := lighthouseDnsService.Spec.ClusterIP
+		if err != nil || lighthouseClusterIp == "" {
 			return goerrors.New("lighthouseDnsService ClusterIp should be available")
 		}
-		expectedCorefile := `#lighthouse
+		coreFile := configMap.Data["Corefile"]
+		if strings.Contains(coreFile, "supercluster.local") {
+			reqLogger.Info("coredns configmap has lighthouse configuration hence updating")
+			if strings.Contains(coreFile, lighthouseClusterIp) {
+				return nil
+			}
+			lines := strings.Split(corefile, "\n")
+			for i, line := range lines {
+				if strings.Contains(line, "supercluster.local") {
+					lines[i+1] = "forward . " + lighthouseClusterIp
+					break
```

Comment:
```suggestion
                                        if strings.Contains(lines[i+1], lighthouseClusterIp) {
				            return nil
			                }
			                
					lines[i+1] = "forward . " + lighthouseClusterIp
					break
```

Then you can remove L375 to avoid a full string scan.